src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -6730,6 +6734,196 @@ public void setIntendedAsReadOnly(List<String> alternatives)
+            dialog.showModal();
+            
+            process.addProcessExitHandler(new ProcessExitEvent.Handler()
+            {
+               @Override
+               public void onProcessExit(ProcessExitEvent event)
+               {
+                  if (event.getExitCode() == 0)
+                  {
+                     success.execute();
+                     dialog.closeDialog();
+                  }
+               }
+            });
+         }
+
+         @Override
+         public void onError(ServerError error)
+         {
+            Debug.logError(error);
+            globalDisplay_.showErrorMessage("Failed to install additional dependencies", error.getUserMessage());
+         }
+      });
+   }
+
+   void checkTestPackageDependencies(final Command success, boolean isTestThat) {
+      dependencyManager_.withTestPackage(
+         new Command()
+         {
+            @Override
+            public void execute()
+            {
+               if (isTestThat)
+                  success.execute();
+               else {
+                  server_.hasShinyTestDependenciesInstalled(new ServerRequestCallback<Boolean>() {
+                     @Override
+                     public void onResponseReceived(Boolean hasPackageDependencies)
+                     {
+                        if (hasPackageDependencies)
+                           success.execute();
+                        else {
+                           globalDisplay_.showYesNoMessage(
+                              GlobalDisplay.MSG_WARNING,
+                              "Install Shinytest Dependencies",
+                              "The package shinytest requires additional system dependencies.\n\n" +
+                              "Install additional system dependencies?",
+                              new Operation()
+                              {
+                                 public void execute()
+                                 {
+                                    installShinyTestDependencies(success);
+                                 }
+                              },
+                              false);
+                        }
+                     }
+                     
+                     @Override
+                     public void onError(ServerError error)
+                     {
+                        Debug.logError(error);
+                        globalDisplay_.showErrorMessage("Failed to check for additional dependencies", error.getMessage());
+                     }
+                  });
+               }
+            }
+         },
+         isTestThat
+      );
+   }
+
+   @Handler
+   void onTestFile()
+   {
+      if (extendedType_ != SourceDocument.XT_TEST_SHINYTEST &&
+          extendedType_ != SourceDocument.XT_TEST_TESTTHAT) {
+         globalDisplay_.showErrorMessage(
+            "Error Running Test",
+            "This file was not recognized as test.");
```

Comment:
It might be better to just disable the command when it can't be used; this is typically what we do for commands which only make sense for certain file types (grep for `DynamicCommands` for examples). What do you think?

Reply:
I think this still needs to be hidden for non tests files, but yes, we can use commands, changed.